crates/pop-parachains/src/build/runtime.rs

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::Error;
+use duct::cmd;
+use pop_common::Profile;
+pub use srtool_lib::{get_image_digest, get_image_tag, ContainerEngine};
+use std::{env, fs, path::PathBuf};
+
+const DEFAULT_IMAGE: &str = "docker.io/paritytech/srtool";
+const ONE_HOUR: u64 = 60 * 60;
+
+/// Builds and executes the command for running the deterministic runtime build process using
+/// srtool.
+pub struct SrToolBuilder {
+	cache_mount: String,
+	default_features: String,
+	digest: String,
+	engine: ContainerEngine,
+	image: String,
+	package: String,
+	path: PathBuf,
+	profile: Profile,
+	runtime_dir: PathBuf,
+	tag: String,
+}
+
+impl SrToolBuilder {
+	/// Creates a new instance of `SrToolBuilder`.
+	///
+	/// # Arguments
+	/// * `engine` - The container engine to use.
+	/// * `path` - The path to the project.
+	/// * `package` - The runtime package name.
+	/// * `profile` - The profile to build the runtime.
+	/// * `runtime_dir` - The directory path where the runtime is located.
+	pub fn new(
+		engine: ContainerEngine,
+		path: Option<PathBuf>,
+		package: String,
+		profile: Profile,
+		runtime_dir: PathBuf,
+	) -> Result<Self, Error> {
+		let default_features = String::new();
+		let tag = get_image_tag(Some(ONE_HOUR)).map_err(|_| Error::ImageTagRetrievalFailed)?;
+		let digest = get_image_digest(DEFAULT_IMAGE, &tag).unwrap_or_default();
+		let dir = fs::canonicalize(path.unwrap_or_else(|| PathBuf::from("./")))?;
+		let tmpdir = env::temp_dir().join("cargo");
+
+		let no_cache = engine == ContainerEngine::Podman;
+		let cache_mount =
+			if !no_cache { format!("-v {}:/cargo-home", tmpdir.display()) } else { String::new() };
+
+		Ok(Self {
+			cache_mount,
+			default_features,
+			digest,
+			engine,
+			image: DEFAULT_IMAGE.to_string(),
+			package,
+			path: dir,
+			profile,
+			runtime_dir,
+			tag,
+		})
+	}
+	/// Executes the runtime build process and returns the path of the generated `.wasm` file.
+	pub fn generate_deterministic_runtime(&self) -> Result<PathBuf, Error> {
```

Comment:
```suggestion
	pub fn build(&self) -> Result<PathBuf, Error> {
```

Hmmm, okay its 'generate' again here when it could just be build.

Reply:
https://github.com/r0gue-io/pop-cli/pull/425/commits/c7dea912f7efe5ac15c68d47e13e6dfc6c2b85cc